src/test_workflow/smoke_test/smoke_test_runner_opensearch.py

```diff
@@ -0,0 +1,81 @@
+from test_workflow.test_args import TestArgs
+from test_workflow.test_result.test_component_results import TestComponentResults
+from test_workflow.test_result.test_result import TestResult
+from test_workflow.test_result.test_suite_results import TestSuiteResults
+
+
+class SmokeTestRunnerOpenSearch(SmokeTestRunner):
+
+    def __init__(self, args: TestArgs, test_manifest: TestManifest) -> None:
+        super().__init__(args, test_manifest)
+        logging.info("Entering Smoke test for OpenSearch Bundle.")
+
+        # TODO: Download the spec from https://github.com/opensearch-project/opensearch-api-specification/releases/download/main-latest/opensearch-openapi.yaml
+        spec_file = os.path.join(os.path.dirname(os.path.abspath(__file__)), "smoke_tests_spec", "opensearch-openapi.yaml")
+        self.spec_ = Spec.from_file_path(spec_file)
+        self.mimetype = {
+            "Content-Type": "application/json"
+        }
+        # self.openapi = openapi_core.OpenAPI.from_file_path(spec_file)
+
+    def validate_request_swagger(self, request: Any) -> None:
+        request = RequestsOpenAPIRequest(request)
+        validate_request(request=request, spec=self.spec_)
+        logging.info("Request is validated.")
+
+    def validate_response_swagger(self, response: Any) -> None:
+        request = RequestsOpenAPIRequest(response.request)
+        response = RequestsOpenAPIResponse(response)
+        validate_response(response=response, spec=self.spec_, request=request)
+        logging.info("Response is validated.")
+
+    def start_test(self, work_dir: Path) -> TestSuiteResults:
+        url = "https://localhost:9200"
+
+        all_results = TestSuiteResults()
+        for component in self.test_manifest.components.select(self.args.components):
+            if component.smoke_test:
+                logging.info(f"Running smoke test on {component.name} component.")
+                component_spec = self.extract_paths_from_yaml(component.name)
+                logging.info(f"component spec is {component_spec}")
+                test_results = TestComponentResults()
+                for api_requests, api_details in component_spec.items():
+                    request_url = ''.join([url, api_requests])
+                    logging.info(f"Validating api request {api_requests}")
+                    logging.info(f"API request URL is {request_url}")
+                    for method in api_details.keys():  # Iterates over each method, e.g., "GET", "POST"
+                        requests_method = getattr(requests, method.lower())
+                        parameters_data = self.convert_parameter_json(api_details.get(method).get("parameters"))
+                        header = api_details.get(method).get("header", self.mimetype)
+                        logging.info(f"Parameter is {parameters_data} and type is {type(parameters_data)}")
+                        logging.info(f"header is {header}")
+                        status = 0
```

Comment:
instead of this, better have try/catch in the called method and return `True/False` and take appropriate action?

Reply:
This status is integer type and used for TestResults class to classify PASS/FAIL.

Reply:
Agree with the recommendation to use boolean instead of integer.

Reply:
The TestResults class will mark PASS only when status == 0. We may have to use integer here.
https://github.com/opensearch-project/opensearch-build/blob/0931ccc2752d9eced6cb3015b822b4a10742fc80/src/test_workflow/test_result/test_result.py#L18-L25